rclcpp_lifecycle/src/transition.cpp

```diff
@@ -99,4 +156,47 @@ Transition::goal_state() const
   return State(transition_handle_->goal);
 }
 
+void
+Transition::reset()
+{
+  // nothing to free here
+  if (!transition_handle_) {
+    owns_rcl_transition_handle_ = false;
+    return;
+  }
+
+  // can't free anything which is not owned
+  if (!owns_rcl_transition_handle_) {
+    transition_handle_ = nullptr;
+    return;
+  }
+
+  if (transition_handle_->start) {
+    if (transition_handle_->start->label) {
+      allocator_.deallocate(
+        const_cast<char *>(transition_handle_->start->label), allocator_.state);
+      transition_handle_->start->label = nullptr;
+    }
+    allocator_.deallocate(transition_handle_->start, allocator_.state);
+    transition_handle_->start = nullptr;
+  }
+  if (transition_handle_->goal) {
+    if (transition_handle_->goal->label) {
+      allocator_.deallocate(
+        const_cast<char *>(transition_handle_->goal->label), allocator_.state);
+      transition_handle_->goal->label = nullptr;
+    }
+    allocator_.deallocate(transition_handle_->goal, allocator_.state);
+    transition_handle_->goal = nullptr;
+  }
+  if (transition_handle_) {
```

Comment:
This condition should always be true when this line is being reached. I would suggest to keep the logic the same as for the `State` class which seems to be correct / clean.